File: src/test/java/javasolutions/arraysandhashing/ValidAnagramTest.java
package javasolutions.arraysandhashing;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ValidAnagramTest {

    @Test
    void should_return_true_if_words_are_anagram() {
        // given
        String firstWord = "anagram";
        String secondWord = "nagaram";
        ValidAnagram validAnagram = new ValidAnagram();
        // when
        boolean firstResult = validAnagram.isAnagramUsingArrays(firstWord, secondWord);
        boolean secondResult = validAnagram.isAnagramUsingMap(firstWord, secondWord);
        // then
        assertAll(
                () -> assertTrue(firstResult),
                () -> assertTrue(secondResult)
        );
    }

    @Test
    void should_return_false_if_words_are_not_anagram() {
        // given
        String firstWord = "rat";
        String secondWord = "car";
        ValidAnagram validAnagram = new ValidAnagram();
        // when
        boolean firstResult = validAnagram.isAnagramUsingArrays(firstWord, secondWord);
        boolean secondResult = validAnagram.isAnagramUsingMap(firstWord, secondWord);
        // then
        assertAll(
                () -> assertFalse(firstResult),
                () -> assertFalse(secondResult)
        );
    }
}

File: src/test/kotlin/kotlinsolutions/arraysandhashing/ValidAnagramTest.kt
package kotlinsolutions.arraysandhashing

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ValidAnagramTest {

    @Test
    fun should_return_true_if_words_are_anagram() {
        // given
        val firstWord = "anagram"
        val secondWord = "nagaram"
        val validAnagram = ValidAnagram()
        // when
        val firstResult = validAnagram.isAnagram(firstWord, secondWord)
        val secondResult = validAnagram.isAnagramInOneLine(firstWord, secondWord)
        val thirdResult = validAnagram.isAnagramUsingHashMap(firstWord, secondWord)
        val fourthResult = validAnagram.isAnagramUsingArray(firstWord,secondWord)
        val fifthResult = validAnagram.isAnagramUsingArraySimplified(firstWord,secondWord)
        // then
        Assertions.assertAll(
            {Assertions.assertTrue(firstResult)},
            {Assertions.assertTrue(secondResult)},
            {Assertions.assertTrue(thirdResult)},
            {Assertions.assertTrue(fourthResult)},
            {Assertions.assertTrue(fifthResult)},
        )
    }

    @Test
    fun should_return_false_if_words_are_not_anagram() {
        // given
        val firstWord = "rat"
        val secondWord = "car"
        val validAnagram = ValidAnagram()
        // when
        val firstResult = validAnagram.isAnagram(firstWord, secondWord)
        val secondResult = validAnagram.isAnagramInOneLine(firstWord, secondWord)
        val thirdResult = validAnagram.isAnagramUsingHashMap(firstWord, secondWord)
        val fourthResult = validAnagram.isAnagramUsingArray(firstWord,secondWord)
        val fifthResult = validAnagram.isAnagramUsingArraySimplified(firstWord,secondWord)
        // then
        Assertions.assertAll(
            {Assertions.assertFalse(firstResult)},
            {Assertions.assertFalse(secondResult)},
            {Assertions.assertFalse(thirdResult)},
            {Assertions.assertFalse(fourthResult)},
            {Assertions.assertFalse(fifthResult)},
        )
    }

}